src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -32,6 +31,10 @@ public static int IndexOf<T>(this Span<T> span, T value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int IndexOf(this Span<byte> span, byte value)
         {
+            if (!BitConverter.IsLittleEndian)
```

Comment:
The locate byte doesn't work on big-endian as it highlights the wrong byte (due to order reversal) https://github.com/aspnet/KestrelHttpServer/pull/1138#issuecomment-255878891; don't know if its worth the effort to come up with something that does work on big-endian though /cc @halter73